File: PHP/NotasDaProvaTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("NotasDaProva.php");

class NotasDaProvaTest extends TestCase {

    public function testNotasDaProva() {
        $obj = new NotasDaProva();
        //Test 1
        $want = "D";
        $got = $obj->takeResult(12);
        $this->assertEquals($want, $got);
        //Test 2
        $want = "A";
        $got = $obj->takeResult(87);
        $this->assertEquals($want, $got);
        //Test 3
        $want = "E";
        $got = $obj->takeResult(0);
        $this->assertEquals($want, $got);
    }

}

File: GO/NotasDaProva_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestNotasDaProva(t *testing.T) {
	tables := []struct {
		top  int
		want string
	}{
		{12, "D"},
		{87, "A"},
		{0, "E"},
	}

	for _, table := range tables {
		got := takeResult(table.top)

		if got != table.want {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}
